src/tour.ts

```diff
@@ -130,6 +178,29 @@ export class Tour extends Evented {
 
     this._setTourID();
 
+    const { apiKey, apiPath } = Shepherd;
+    // If we have an API key, then setup Pro features
+    if (apiKey && apiPath) {
+      this.dataRequester = new DataRequest(apiKey, apiPath);
+
+      const shepherdProId = localStorage.getItem('shepherdPro:userId');
+      this.currentUserId = shepherdProId;
+
+      this.trackedEvents.forEach((event) =>
+        this.on(event, (opts: Record<string, unknown>) => {
+          const { tour } = opts;
+          const { id, steps } = tour as Tour;
+
+          const data = {
+            currentUserId: this.currentUserId,
+            eventType: event,
+            tour: { id, numberOfSteps: steps.length }
+          };
+          if (this.dataRequester) this.dataRequester.sendEvents({ data });
```

Comment:
```suggestion
          this.dataRequester?.sendEvents({ data });
```

Just a nitpicky suggestion that we could use optional chaining here